rust/rsc/src/rsc/blob_store_impls.rs

```diff
@@ -75,3 +85,56 @@ impl BlobStore for TestBlobStore {
 }
 
 impl DebugBlobStore for TestBlobStore {}
+
+#[derive(Debug, Clone)]
+pub struct DbOnlyBlobStore {
+    pub id: Uuid,
+}
+
+#[async_trait]
+impl BlobStore for DbOnlyBlobStore {
+    fn id(&self) -> Uuid {
+        self.id
+    }
+
+    async fn stream<'a>(
+        &self,
+        stream: BoxStream<'a, Result<Bytes, std::io::Error>>,
+    ) -> Result<String, std::io::Error> {
+        let reader = StreamReader::new(stream);
+        futures::pin_mut!(reader);
+        let mut buffer = [0u8; 101];
+        let mut count = 0;
+
+        loop {
+            // If we have filled the buffer then the request has exceeded the max.
+            if count == 101 {
```

Comment:
Maybe this isn't a problem with your invocation of `reader.read` but I would expect this to be `>=` in case we read say 200 bytes in the first go.

Reply:
Happy to change it but `reader.read` uses the type to ensure it never overflows the buffer so the count can never go past `101`

Reply:
Fancy. No need to change it then.